
Pool shape : torch.Size([40000, 1, 28, 28])
[run_image_classification_40k_epig_acq.py:166]INFO: Initial test log-likelihood: -4.4428
[run_image_classification_40k_epig_acq.py:167]INFO: Initial test Bayes log-likelihood: -2.1742
[run_image_classification_40k_epig_acq.py:168]INFO: Initial accuracy: 48.72%
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
Starting epig acquisition
Using original estimate_epig function
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
[utils.py:19]WARNING: Invalid EPIG score (min = nan, max = nan)
Traceback (most recent call last):
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 67, in estimate_epig_minibatch
    features = m_model.la.predictive_samples(combined_inputs, n_samples=150)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 660, in predictive_samples
    f_mu, f_var = self._glm_predictive_distribution(x)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 674, in _glm_predictive_distribution
    Js, f_mu = self.backend.jacobians(X, enable_backprop=self.enable_backprop)
  File "/usr/local/lib/python3.10/dist-packages/laplace/curvature/asdl.py", line 44, in jacobians
    f = batch_gradient(self.model, loss_fn, x, None).detach()
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/gradient.py", line 58, in batch_gradient
    f = model(inputs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/container.py", line 215, in forward
    input = module(input)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1581, in _call_impl
    hook_result = hook(self, args, result)
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/core.py", line 21, in forward_hook
    in_data = _preprocess_in_data(module, in_data, out_data)
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/core.py", line 68, in _preprocess_in_data
    in_data = im2col_2d(in_data, module)
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/utils.py", line 53, in im2col_2d
    Mx = F.unfold(
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/functional.py", line 4756, in unfold
    return torch._C._nn.im2col(input, _pair(kernel_size), _pair(dilation), _pair(padding), _pair(stride))
KeyboardInterrupt
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 275, in <module>
    main(**args)
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 193, in main
    scores = estimate_epig(learner, dataset.get_pool_loader(batch_size=batch_size), target_inputs, cfg.acquisition.epig_using_matmul)
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 51, in estimate_epig
    epig_scores = estimate_epig_minibatch(m_model, inputs, target_inputs, use_matmul)  # [B,]
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 72, in estimate_epig_minibatch
    features = m_model.la.predictive_samples(combined_inputs, n_samples=150, diagonal_output = True)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 660, in predictive_samples
    f_mu, f_var = self._glm_predictive_distribution(x)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 674, in _glm_predictive_distribution
    Js, f_mu = self.backend.jacobians(X, enable_backprop=self.enable_backprop)
  File "/usr/local/lib/python3.10/dist-packages/laplace/curvature/asdl.py", line 44, in jacobians
    f = batch_gradient(self.model, loss_fn, x, None).detach()
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/gradient.py", line 60, in batch_gradient
    loss.backward()
  File "/usr/local/lib/python3.10/dist-packages/torch/_tensor.py", line 492, in backward
    torch.autograd.backward(
  File "/usr/local/lib/python3.10/dist-packages/torch/autograd/__init__.py", line 251, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
KeyboardInterrupt
Traceback (most recent call last):
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 67, in estimate_epig_minibatch
    features = m_model.la.predictive_samples(combined_inputs, n_samples=150)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 660, in predictive_samples
    f_mu, f_var = self._glm_predictive_distribution(x)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 674, in _glm_predictive_distribution
    Js, f_mu = self.backend.jacobians(X, enable_backprop=self.enable_backprop)
  File "/usr/local/lib/python3.10/dist-packages/laplace/curvature/asdl.py", line 44, in jacobians
    f = batch_gradient(self.model, loss_fn, x, None).detach()
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/gradient.py", line 58, in batch_gradient
    f = model(inputs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/container.py", line 215, in forward
    input = module(input)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1581, in _call_impl
    hook_result = hook(self, args, result)
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/core.py", line 21, in forward_hook
    in_data = _preprocess_in_data(module, in_data, out_data)
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/core.py", line 68, in _preprocess_in_data
    in_data = im2col_2d(in_data, module)
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/utils.py", line 53, in im2col_2d
    Mx = F.unfold(
  File "/usr/local/lib/python3.10/dist-packages/torch/nn/functional.py", line 4756, in unfold
    return torch._C._nn.im2col(input, _pair(kernel_size), _pair(dilation), _pair(padding), _pair(stride))
KeyboardInterrupt
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 275, in <module>
    main(**args)
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 193, in main
    scores = estimate_epig(learner, dataset.get_pool_loader(batch_size=batch_size), target_inputs, cfg.acquisition.epig_using_matmul)
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 51, in estimate_epig
    epig_scores = estimate_epig_minibatch(m_model, inputs, target_inputs, use_matmul)  # [B,]
  File "/dss/dsshome1/0D/ge32jaq2/epig_plus_orig_marglik/active_learning_marglik/run_image_classification_40k_epig_acq.py", line 72, in estimate_epig_minibatch
    features = m_model.la.predictive_samples(combined_inputs, n_samples=150, diagonal_output = True)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 660, in predictive_samples
    f_mu, f_var = self._glm_predictive_distribution(x)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/laplace/baselaplace.py", line 674, in _glm_predictive_distribution
    Js, f_mu = self.backend.jacobians(X, enable_backprop=self.enable_backprop)
  File "/usr/local/lib/python3.10/dist-packages/laplace/curvature/asdl.py", line 44, in jacobians
    f = batch_gradient(self.model, loss_fn, x, None).detach()
  File "/usr/local/lib/python3.10/dist-packages/asdfghjkl/gradient.py", line 60, in batch_gradient
    loss.backward()
  File "/usr/local/lib/python3.10/dist-packages/torch/_tensor.py", line 492, in backward
    torch.autograd.backward(
  File "/usr/local/lib/python3.10/dist-packages/torch/autograd/__init__.py", line 251, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
KeyboardInterrupt